sdk/eventhub/event-hubs/src/eventhubSharedKeyCredential.ts

```diff
@@ -78,16 +78,12 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<{
-      SharedAccessSignature: string;
-      SharedAccessKeyName: string;
-      SharedAccessKey: string;
-    }>(connectionString);
+    const parsed = parseEventHubConnectionString(connectionString);
 
-    if (parsed.SharedAccessSignature == null) {
-      return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    if (parsed.sharedAccessSignature) {
```

Comment:
Just curious.. what was the reason to flip the if/else check here?

Reply:
It was a while ago 😄  but I think my thought process at the time was to use truthyness to cover `undefined`, `null`, and empty strings in one conditional

Reply:
From what I recall when I wrote the same for Service Bus, I was trying to pivot on the existence of the key name in the connection string rather than pivot on the key value pushing the responsibility of validating the actual value to the `SharedKeyCredential` class. Now I see that if I really wanted to achieve that purpose I should have done something like `parsed.hasOwnProperty("sharedAccessSignature")`. Does `parseEventHubConnectionString` explicity set `sharedAccessSignature` to null or undefined when it doesnt find it in the connection string or not set it at all?

Reply:
Right now it doesn't set it all. I remember going back and forth about it but deciding to not set the key at all in case someone pivots on the presence of the key instead of the presence of the value (like you suggested here). Do you think it makes more sense to check if that property is set at all instead of whether the value exists and is not empty?

Reply:
I would prefer this if/else check to pivot on presence of key name and then have the SharedKeyCredential class to validate the value